File: Problems/Algorithms/49. Group Anagrams/GroupAnagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        List<List<String>> ans = new ArrayList<>();
        Map<String, List<String>> map = new HashMap<>();
        
        for (String str : strs) {
            char[] charArr = str.toCharArray();
            Arrays.sort(charArr);
            String curr = String.valueOf(charArr);
            if (map.containsKey(curr)) {
                map.get(curr).add(str);
            } else {
                map.put(curr, new ArrayList<String>());
                map.get(curr).add(str);
            }
        }
        
        for (String key : map.keySet()) {
            ans.add(map.get(key));
        }
        
        return ans;
    }
}

File: Problems/Algorithms/49. Group Anagrams/GroupAnagrams.kt
class Solution {
    fun groupAnagrams(strs: Array<String>): List<List<String>> {
        val ans = mutableListOf<MutableList<String>>()
        val map = mutableMapOf<String, MutableList<String>>()
        
        for (string in strs) {
            val curr = string.toCharArray().sorted().joinToString()
            if (map.containsKey(curr)) {
                map[curr]!!.add(string)
            } else {
                map[curr] = mutableListOf<String>()
                map[curr]!!.add(string)
            }
        }
        
        for (key in map.keys) {
            ans.add(map[key]!!)
        }
        
        return ans
    }
}

